paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -261,51 +261,95 @@ def moving_average_forecast_policy(historical_load, moving_average_window_second
     return total / count
 
 
-HISTORICAL_LOAD_SERIALIZATION_FORMAT = 'dd'
-SIZE_PER_HISTORICAL_LOAD_RECORD = struct.calcsize(HISTORICAL_LOAD_SERIALIZATION_FORMAT)
+HISTORICAL_LOAD_SERIALIZATION_FORMAT_v0 = 'dd'
+SIZE_PER_HISTORICAL_LOAD_RECORD_v0 = struct.calcsize(HISTORICAL_LOAD_SERIALIZATION_FORMAT_v0)
 
 
-def zk_historical_load_path(zk_path_prefix):
-    return "%s/historical_load" % zk_path_prefix
+def zk_historical_load_path(zk_path_prefix, version=1):
+    if version:
+        return "%s/historical_load_v%d" % (zk_path_prefix, version)
+    else:
+        return "%s/historical_load" % zk_path_prefix
 
 
-def save_historical_load(historical_load, zk_path_prefix):
+def save_historical_load(historical_load, zk_path_prefix, version=1):
     with ZookeeperPool() as zk:
-        historical_load_bytes = serialize_historical_load(historical_load)
-        zk.ensure_path(zk_historical_load_path(zk_path_prefix))
-        zk.set(zk_historical_load_path(zk_path_prefix), historical_load_bytes)
+        historical_load_bytes = serialize_historical_load(historical_load, version=version)
+        zk.ensure_path(zk_historical_load_path(zk_path_prefix, version=version))
+        zk.set(zk_historical_load_path(zk_path_prefix, version=version), historical_load_bytes)
 
 
-def serialize_historical_load(historical_load):
-    max_records = 1000000 // SIZE_PER_HISTORICAL_LOAD_RECORD
+def serialize_historical_load_v0(historical_load):
+    max_records = 1000000 // SIZE_PER_HISTORICAL_LOAD_RECORD_v0
     historical_load = historical_load[-max_records:]
-    return b''.join([struct.pack(HISTORICAL_LOAD_SERIALIZATION_FORMAT, *x) for x in historical_load])
+    return b''.join([struct.pack(HISTORICAL_LOAD_SERIALIZATION_FORMAT_v0, *x) for x in historical_load])
+
+
+def serialize_historical_load_v1(historical_load, max_size=1000000):
+    historical_load_bytes = b'\n'.join([b"%d %.2f" % x for x in historical_load])
+
+    # trim to max size
+    historical_load_bytes_trimmed = (b'\n' + historical_load_bytes)[-(max_size + 1):]
+    first_newline = historical_load_bytes_trimmed.find('\n')
+    if first_newline == -1:
+        return b''
+    else:
+        return historical_load_bytes_trimmed[first_newline + 1:]
 
 
-def fetch_historical_load(zk_path_prefix):
+def fetch_historical_load(zk_path_prefix, versions=(0, 1)):
     with ZookeeperPool() as zk:
-        try:
-            historical_load_bytes, _ = zk.get(zk_historical_load_path(zk_path_prefix))
-            return deserialize_historical_load(historical_load_bytes)
-        except NoNodeError:
-            return []
+        historical_load = []
+        for version in versions:
+            try:
+                historical_load_bytes, _ = zk.get(zk_historical_load_path(zk_path_prefix, version=version))
+                historical_load.extend(deserialize_historical_load(historical_load_bytes, version=version))
+            except NoNodeError:
+                pass
+        return sorted(historical_load)
```

Comment:
Are you sure it is ok for this list to have multiple versions inside of it? Can the consumers tolerate that?